### PR TITLE
Remove unused snapshots

### PR DESCRIPTION
**Pull request by harupy**
How I found unused snapshots:

```
cargo insta test --delete-unreferenced-snapshots
```

It'd be nice if we could detect them in CI.

### CODE REVIEW

**Comment by harupy** [comment]
Note this PR does remove `src/pyflakes/snapshots/ruff__pyflakes__tests__F831_F831.py.snap` as https://github.com/charliermarsh/ruff/pull/1496 does.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -117,7 +117,15 @@ jobs:
...
-      - run: cargo test --all
+      - run: |
+          export INSTA_SNAPSHOT_REFERENCES_FILE="$(mktemp)"
+          cargo test --all
+          UNUSED_SNAPSHOTS=$(cat <(git ls-files '*.snap' | xargs realpath) <(cat $INSTA_SNAPSHOT_REFERENCES_FILE) | sort | uniq -u)
+          if [ ! -z "$UNUSED_SNAPSHOTS" ]; then
+            echo "Found unused snapshots:"
+            echo "$UNUSED_SNAPSHOTS"
+            exit 1
+          fi
```
Comment (harupy) [inline .github/workflows/ci.yaml:128]: When unused snapshots exist:

https://github.com/charliermarsh/ruff/actions/runs/3811933348/jobs/6484869555

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -117,7 +117,15 @@ jobs:
             ${{ runner.os }}-build-
             ${{ runner.os }}-
       - run: pip install black[d]==22.12.0
-      - run: cargo test --all
+      - run: |
+          export INSTA_SNAPSHOT_REFERENCES_FILE="$(mktemp)"
```
Comment (harupy) [inline .github/workflows/ci.yaml:121]: Referecne: https://docs.rs/insta/1.3.0/insta/#deleting-unused-snapshots

**Comment by charliermarsh** [comment]
Thank you :)

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (charliermarsh) [reply .github/workflows/ci.yaml]: The other option would be to run `cargo insta test --delete-unreferenced-snapshots` then `git diff`... but that requires running tests twice.

Comment (max-sixty) [reply .github/workflows/ci.yaml]: Forgive the drive-by but I came across this — another approach is running `cargo insta test --delete-unreferenced-snapshots` as the main test command, and then doing `git diff` — then tests only need to run once. And `cargo insta test` covers all tests.

Comment (harupy) [reply .github/workflows/ci.yaml]: @max-sixty Thanks for the comment. I thought `cargo insta test` only runs snapshot tests but looks like I was wrong. I'll file a RP :)

**Pull request merged**